docs/adding_a_dataset.md

```diff
@@ -0,0 +1,252 @@
+            labels.extend(_label)
+
+            assert len(documents) == len(labels)
+
+            rng = random.Random(1111)  # local only seed
+            pairs = list(zip(documents, labels))
+            rng.shuffle(pairs)
+            documents, labels = [list(collection) for collection in zip(*pairs)]
+
+            # To get a more robust estimate we create batches of size 512, this decision can vary depending on dataset
+            documents_batched = list(batched(documents, 512))
+            labels_batched = list(batched(labels, 512))
+
+            # reduce the size of the dataset as we see that we obtain a consistent scores (if we change the seed) even
+            # with only 512x4 samples. 
+            documents_batched = documents_batched[:4]
+            labels_batched = labels_batched[:4]
+
+
+            ds[split] = datasets.Dataset.from_dict(
+                {
+                    "sentences": documents_batched,
+                    "labels": labels_batched,
+                }
+            )
+
+        self.dataset = datasets.DatasetDict(ds)
+```
+
+</details>
+
+
+## 2) Creating the metadata object
+Along with the task MTEB requires metadata regarding the task. If the metadata isn't available please provide your best guess or leave the field as `None`.
+
+To get an overview of the fields in the metadata object, you can look at the [TaskMetadata](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/TaskMetadata.py) class.
+
+
+Note that these fields can be left blank if the information is not available and can be extended if necessary. We do not include any machine-translated (without verification) datasets in the benchmark.
+
+<details closed>
+<summary>Domains</summary>
+<br>
+
+The domains follow the categories used in the [Universal Dependencies project](https://universaldependencies.org), though we updated them where deemed appropriate. These do not have to be mutually exclusive.
+
+| **Domain**    | **Description**                                                  |
+| ------------- | ---------------------------------------------------------------- |
+| Academic      | Academic writing                                                 |
+| Religious     | Religious text e.g. bibles                                       |
+| Blog          | [Blogpost, weblog etc.](https://en.wikipedia.org/wiki/Blog)      |
+| Fiction       | Works of [fiction](https://en.wikipedia.org/wiki/Fiction)        |
+| Government    | Governmental communication, websites or similar                  |
+| Legal         | Legal documents, laws etc.                                       |
+| Medical       | doctors notes, medical procedures or similar                     |
+| News          | News articles, tabloids etc.                                     |
+| Reviews       | Reviews e.g. user reviews of products                            |
+| Non-fiction   | [non-fiction](https://en.wikipedia.org/wiki/Non-fiction) writing |
```

Comment:
This category overlaps quite a bit with the others. Maybe we should specify that this is for "material which does not fit any other category"?